validation/superchain-version_test.go

```diff
@@ -80,69 +57,107 @@ func TestSuperchainWideContractVersions(t *testing.T) {
-		919:       true, // sepolia/mode   L1CrossDomainMessengerProxy.version=1.4.1, https://github.com/ethereum-optimism/security-pod/issues/105
-		1740:      true, // sepolia/metal  L1CrossDomainMessengerProxy.version=1.4.1, https://github.com/ethereum-optimism/security-pod/issues/105
-		1750:      true, // mainnet/metal  L1CrossDomainMessengerProxy.version=1.4.1, https://github.com/ethereum-optimism/security-pod/issues/105
-		8453:      true, // mainnet/base
-		8866:      true, // mainnet/superlumio L1CrossDomainMessengerProxy.version=1.4.1, https://github.com/ethereum-optimism/security-pod/issues/105
-		34443:     true, // mainnet/mode
-		84532:     true, // sepolia/base
-		90001:     true, // sepolia/race, due to https://github.com/ethereum-optimism/superchain-registry/issues/147
-		7777777:   true, // mainnet/zora
-		11155421:  true, // sepolia-dev-0/oplabs-devnet-0
-		999999999: true, // sepolia/zora
-	}
-
-	checkOPChainSatisfiesSemver := func(t *testing.T, chain *ChainConfig) {
-		// TODO: don't hardcode this https://github.com/ethereum-optimism/superchain-registry/issues/219
-		isFaultProofChain := map[uint64]bool{
-			10:       true, // OP Mainnet
-			11155420: true, // OP Sepolia
-		}
+		8453:     true, // mainnet/base
+		11155421: true, // sepolia-dev-0/oplabs-devnet-0
+		11763072: true, // sepolia-dev-0/base-devnet-0
+	}
 
+	checkOPChainMatchesATag := func(t *testing.T, chain *ChainConfig) {
 		rpcEndpoint := Superchains[chain.Superchain].Config.L1.PublicRPC
-
 		require.NotEmpty(t, rpcEndpoint)
 
 		client, err := ethclient.Dial(rpcEndpoint)
 		require.NoErrorf(t, err, "could not dial rpc endpoint %s", rpcEndpoint)
-
-		contractNames := BaseStackContractNames
-		if isFaultProofChain[chain.ChainID] {
-			// The FaultProof contracts are only present in chains that have activated fault proofs
-			contractNames = append(contractNames, FaultProofContractNames...)
-		} else {
-			// The L2OutputOracle contract is only present in chains that have not activated fault proofs
-			contractNames = append(contractNames, "L2OutputOracle")
-		}
-
-		for _, contractName := range contractNames {
-			desiredSemver, err := SuperchainSemver[chain.Superchain].VersionFor(contractName)
-			require.NoError(t, err)
-			// ASSUMPTION: we will check the version of the implementation via the declared proxy contract
-			var localizedContractName string
-			if slices.Contains(UnproxiedContractNames, contractName) {
-				localizedContractName = contractName
-			} else {
-				localizedContractName = contractName + "Proxy"
-			}
-			contractAddress, err := Addresses[chain.ChainID].AddressFor(localizedContractName)
-			require.NoErrorf(t, err, "%s/%s.%s.version= UNSPECIFIED", chain.Superchain, chain.Name, localizedContractName)
-			checkSemverForContract(t, localizedContractName, &contractAddress, client, desiredSemver)
-		}
+		isFPAC := chain.ChainID == 10 || chain.ChainID == 11155420 || chain.ChainID == 11155421 // TODO don't hardcode this
```

Comment:
This is still TODO.

Reply:
Request from product that we make the fault proofyness of each chain a blindingly obvious human readable string.

Reply:
[f9e61bd](https://github.com/ethereum-optimism/superchain-registry/pull/297/commits/f9e61bdffbec53a6a3d73aaf78f36af210b2b8c3)